tools/project_manager/page_compare.php

```diff
@@ -0,0 +1,213 @@
+<?php
+$relPath="./../../pinc/";
+include_once($relPath.'base.inc');
+include_once($relPath.'misc.inc');
+include_once($relPath.'theme.inc');
+include_once($relPath.'Project.inc');
+include_once($relPath."DPage.inc"); // remove_formatting()
+
+require_login();
+
+$comparator = new Comparator();
+$comparator->get_data();
+$comparator->render();
+
+class Comparator
+{
+    function __construct()
+    {
+        global $PROJECT_STATES_IN_ORDER;
+
+        $this->L_round_options = array('P1', 'P2', 'P3', 'F1');
+        $this->R_round_options = array('F1', 'F2');
+        $this->state_index = array_flip($PROJECT_STATES_IN_ORDER);
+    }
+
+    function get_data()
+    {
+        $this->projectid = array_get($_GET, 'project', null);
+        $this->L_round_id = get_enumerated_param($_GET, "L_round_id", "P3", $this->L_round_options);
+        $this->R_round_id = get_enumerated_param($_GET, "R_round_id", "F1", $this->R_round_options);
+        $this->page_set = get_enumerated_param($_GET, "page_set", "all", array('left', 'right', 'all'));
+        $this->go_compare = isset($_GET['compare']);
+    }
+
+    function render()
+    {
+        global $pguser, $code_url;
+
+        $title = _('Compare pages with formatting removed');
+        output_header("$title", NO_STATSBAR);
+        echo "<h1>$title</h1>\n";
+
+        $error_message = "";
+        if($this->projectid)
+        {
+            // this function exits if invalid
+            $this->projectid = validate_projectID('project', $this->projectid);
+        }
+        else
+        {
+            $error_message = _("No project is specified");
+        }
+
+        if($this->projectid)
+        {
+            try
+            {
+                $this->project = new Project($this->projectid);
```

Comment:
You don't need the try/catch. On the TEST server when we throw exceptions we print the full exception to help debugging. On PROD we catch the exception in a global exception handler and just print the error. See the top of `pinc/base.inc` for the code that does this.